protocol/communities/manager.go

```diff
@@ -1650,6 +1650,11 @@ func (m *Manager) EditCommunity(request *requests.EditCommunity) (*Community, er
 
 	if community.IsControlNode() {
 		community.increaseClock()
+		// Update the community cache. This makes sure that the image server has access to the new image if needed
+		_, _, err = m.preprocessDescription(request.CommunityID, community.config.CommunityDescription)
```

Comment:
This doesn't seem quite right. First, the community is edited, followed by the generation of a new description. Subsequently, the new description undergoes reprocessing, which involves the complete mechanism of decryption and populating encryption-specific cache tables. In this case, the cache is used to store _unencrypted_ descriptions for subsequent processing. However, the invariant is broken in this scenario, and I’m not sure how this ties into the image server.

Reply:
It took me a while to understand what was going on. The image server relies on the community cache to serve the images, ie `encrypted_community_description_cache`.

When calling `EditCommunity` the image does get saved in the `communities` DB. However, the cache does **not** get updated straight away.
From what i could gather, it gets updated the next time `getByID` gets called, which is just a race condition at this point.

Calling `preprocessDescription` after editing the clock makes sure the new description gets saved in the cache DB.

Maybe I could call `SaveDecryptedCommunityDescription` directly instead of `preprocessDescription`? It would be clearer in the code, but `preprocessDescription` does a bit more handling with regards to members, so I thought it would be safer to use that instead.

@osmaczko do you have a suggestion on how to make it better? Maybe a better comment?

Reply:
> @jrainville would you mind fixing that within the scope of this PR please?

I don't mind. I'll just wait for @qfrank 's reply to make sure there isn't something we don,t know about. Maybe the cache serves faster?

Reply:
I understand the reasoning. However, `encrypted_community_description_cache` should not be used to retrieve the community description, as it can be invalidated at any time. This means the entry might be removed and become inaccessible. Moreover, `GetByID` should be used instead because it applies events on top of the raw community description. With the current implementation, if an admin changes the image (logo or banner) and the change is not yet approved by the control node, then the image will not be read properly by the media server.

Reply:
Fixed here: https://github.com/status-im/status-go/pull/6127

I'll rebase this PR on top of that other one